physics/mp_thompson.F90

```diff
@@ -260,6 +272,13 @@ subroutine mp_thompson_init(ncol, nlev, con_g, con_rd, con_eps,   &
 
            ! Copy to local array for calculating cloud effective radii below
            nc_local = nc
+ 
+        else if (merra2_aerosol_aware) then
+
+           ! Ensure we have 1st guess cloud droplet number where mass non-zero but no number.
+           where(qc .LE. 0.0) nc=0.0
+           where(qc .GT. 0 .and. nc .LE. 0.0) nc = make_DropletNumber(qc*rho, nwfa*rho) * orho
```

Comment:
A new comment here about having all the other where statements using floating point 0.0 but the qc test is using simply the number zero without a floating point representation.  Shouldn't they all be consistent?  It also concerns me that this setting to zero ONLY appears to occur for MERRA aerosol so why is it not done for the previous?